t/dispatch.t

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env perl
+
+use 5.028;
+use warnings;
+use experimental qw(signatures);
+
+use Test::Lib;
+use Test::PromAlertProxy;
+use Test::More;
+use Test::Time time => Test::PromAlertProxy->now;
+
+use IO::Async::Loop;
+
+use PromAlertProxy::Hub;
+use PromAlertProxy::PromAlert;
+
+use PromAlertProxy::Logger '$Logger' => { init => {
+  ident    => 'promalertproxy',
+  to_self  => 1,
+} };
+
+my $loop = IO::Async::Loop->new;
+
+my $hub = PromAlertProxy::Hub->new(loop => $loop);
+
+my $target1 = Test::PromAlertProxy::Target->new(
+  hub     => $hub,
+  id      => 'test1',
+  default => 1,
+);
+my $target2 = Test::PromAlertProxy::Target->new(
+  hub     => $hub,
+  id      => 'test2',
+);
+my $crashtarget = Test::PromAlertProxy::CrashTarget->new(
```

Comment:
I wonder, would it be cool to do these from config instead? (And then you'd also be testing the config guts, which aren't tested right now.) It would also be less line noise in this test.

Reply:
Yeah, probably right. I'll tweak config a little to take a string. Lemme see where we land.

Reply:
Initial thought: https://github.com/robn/promalertproxy/commit/v2-config-test

Nice: being able to write config

Unnice: neading to squirrel around in an internal attribute to get the target handles out

Extraunnice: doesn't actually run, because the server config is expected but doesn't exist. Which is always a little awkward in tests (not actually the thing under test).

This might be less of a thing if the whole thing was structured differently as we've discussed, but at this point its not clear to me that that's an improvement.

I will at least write a separate config test though.

Reply:
> Extraunnice: doesn't actually run, because the server config is expected but doesn't exist. Which is always a little awkward in tests (not actually the thing under test).

That feels like an artifact of the fact that hub and config can't decide what's in charge. Server IP/port are required in config, but not on the hub, even though the hub requires them to do anything useful.

But anyway, in tests I like Net::EmptyPort, and then you can just write `server.ip = "localhost", server.port = empty_port()` (or moral equivalent).